src/app-layer-htp.c

```diff
@@ -4282,64 +3996,53 @@ libhtp:\n\
     addr = "192.168.10.42";
     if (inet_pton(AF_INET, addr, buf) == 1) {
         (void)SCRadixFindKeyIPV4BestMatch(buf, cfgtree, &user_data);
-        if (user_data != NULL) {
-            HTPCfgRec *htp_cfg_rec = user_data;
-            htp = htp_cfg_rec->cfg;
-            SCLogDebug("LIBHTP using config: %p", htp);
-        }
-        if (htp == NULL) {
-            printf("Could not get config for: %s\n", addr);
-            goto end;
-        }
+        FAIL_IF_NULL(user_data);
+        HTPCfgRec *htp_cfg_rec = user_data;
+        htp = htp_cfg_rec->cfg;
+        SCLogDebug("LIBHTP using config: %p", htp);
+
+        FAIL_IF_NULL(htp);
+
     }
```

Comment:
I wonder if we can simplify this to just check:
```
FAIL_IF_NOT(inet_pton(AF_INET, addr, buf) == 1)
```
since the else case will simply be a `FAIL`. What do you think, @inashivb ?

Reply:
I think we need to check user data and htp config too. So, yes to your suggestion, followed by the rest of the matches too. The block could look like:


```C
        FAIL_IF(inet_pton(AF_INET, addr, buf) != 1)
        (void)SCRadixFindKeyIPV4BestMatch(buf, cfgtree, &user_data);
        FAIL_IF_NULL(user_data);
        HTPCfgRec *htp_cfg_rec = user_data;
        htp = htp_cfg_rec->cfg;
        SCLogDebug("LIBHTP using config: %p", htp);

        FAIL_IF_NULL(htp);
```

Reply:
yeah this was my first choice but then I was trying to make it replicate the current code.. I guess this works though. Let's do this! :)